pallets/rmrk-core/src/functions.rs

```diff
@@ -270,8 +270,8 @@ where
 		_sender: T::AccountId,
 		owner: T::AccountId,
 		collection_id: CollectionId,
-		recipient: Option<T::AccountId>,
-		royalty: Option<Permill>,
+		royalty_recipient: Option<T::AccountId>,
```

Comment:
Couldn't we instead do this?
```
royalty: Option<RoyaltyInfo<T::AccountId>>
```
Removes the need for the line 285-296 I believe.